homeassistant/components/wled/coordinator.py

```diff
@@ -33,9 +38,69 @@ def update_listeners(self) -> None:
         for update_callback in self._listeners:
             update_callback()
 
+    async def _use_websocket(self) -> None:
+        """Use WebSocket for updates, instead of polling."""
+        if self.wled.connected:
+            # We are already connected
+            return
+
+        # Connect to WebSocket
+        try:
+            await self.wled.connect()
+        except WLEDError as err:
+            self.logger.info(err)
+            return
+
+        async def close_websocket(_) -> None:
+            """Close WebSocket connection."""
+            await self.wled.disconnect()
+
+        # Clean disconnect WebSocket on Home Assistant shutdown
+        self.unsub = self.hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_STOP, close_websocket
+        )
+
+        def update_data_from_websocket(device: WLEDDevice) -> None:
+            """Call when WLED reports a state change."""
+            self.data = device
+            self.update_listeners()
+
+        async def listen() -> None:
+            """Listen for state changes via WebSocket."""
+            # Disable polling
+            self.update_interval = None
+
+            try:
+                await self.wled.listen(callback=update_data_from_websocket)
+            except WLEDConnectionClosed as err:
+                self.last_update_success = False
+                self.logger.info(err)
+            except WLEDError as err:
+                self.last_update_success = False
+                self.logger.error(err)
+
+            # Ensure we are disconnected
+            await self.wled.disconnect()
+            if self.unsub:
+                self.unsub()
+                self.unsub = None
+
+            # Resume polling
```

Comment:
Comment uses the word "resume" but I don't see where you suspend the polling?

Reply:
oh see it now inside `listen`.

Reply:
- The WebSocket can be triggered by: `_async_update_data`, which is a scheduled refresh / first time triggered update.
- In the first line of `listen()` is sets `self.update_interval = None` causing the next schedule not to happen.

Reply:
Removed that part, as not all state changes cause a push of data over the WebSocket. The data is available but only pushed when some other event that does trigger WebSocket updates is triggered.

So I'll keep polling around, at a slower rate. Expected that most devices will support WebSockets anyway. WebSockets are around in a hidden/experimental state since August 2020, and the integration will actually work with those ancient versions as well.